localstack/services/apigateway/invocations.py

```diff
@@ -234,7 +243,8 @@ def invoke_rest_api(invocation_context: ApiInvocationContext):
         return make_error_response("Invalid request body", 400)
 
     api_key_required = resource.get("resourceMethods", {}).get(method, {}).get("apiKeyRequired")
-    if not is_api_key_valid(api_key_required, headers, invocation_context.stage):
+    if api_key_required and not is_api_key_valid(invocation_context):
```

Comment:
According to https://repost.aws/knowledge-center/api-gateway-troubleshoot-403-forbidden, the message would be "Invalid API Key identifier specified"

Reply:
Yes, that is weird because from my tests in AWS, I always only got a json message with `{"message": "Forbidden"}` and nothing else